tests/acceptance/form-rehydration-test.js

```diff
@@ -43,4 +43,64 @@ test('it restores focus', async function(assert) {
   assert.equal(find('input[name="title"]'), document.activeElement);
 
   await destroyApp(this.application);
-});
\ No newline at end of file
+});
+
+test('it restores selection for the active element', async function(assert) {
+  document.querySelector(rootSelector).innerHTML = staticFormMarkup;
+  let input = document.querySelector(`${rootSelector} input[name="title"]`);
+  input.value = 'foo';
+  input.focus();
+
+  input.setSelectionRange(0, 1); // Selecting "f" from "foo"
+
+  this.application = startApp();
+
+  await visit('/');
+
+  let resultInput = find('input[name="title"]');
+  assert.equal(resultInput.selectionStart, 0, 'selectionStart is correct');
+  assert.equal(resultInput.selectionEnd, 1, 'selectionEnd is correct');
+
+  await destroyApp(this.application);
+});
+
+// https://html.spec.whatwg.org/multipage/input.html#concept-input-apply
+test(`it doesn't throw for an input with no selection support`, async function(assert) {
+  assert.expect(0);
+
+  document.querySelector(rootSelector).innerHTML = `<input name="title" type="number">`;
+  document.querySelector(`${rootSelector} input[name="title"]`).focus();
+
+  this.application = startApp();
+
+  await visit('/');
+  await destroyApp(this.application);
+});
+
+
+// All input selections could be restored in theory, but it adds very little value in practice...
+skip(`it restores selection for all input elements`, async function(assert) {
```

Comment:
Tbh, I wasn't even aware that non-focused elements can have a selection, is that really the case? Do you think this should be supported (in the future)? If not, we should remove this test I think! Would stay around being skipped forever ;)

Reply:
> Tbh, I wasn't even aware that non-focused elements can have a selection, is that really the case?

Yes, for example when navigating around inputs using the tab key on the keyboard, all individual input's selection/caret position and is retained.

> Do you think this should be supported (in the future)?

Even if it's supported, one needs to have a reaaaally long running promise or slow loading JS assets to benefit, but even if that's the case, I think it's hardly noticable. What I was thinking while writing the test is that since browsers normally support this feature and the whole rehydration process should be as seamless as possible in theory, it's OK to mention this capability - yet, I don't want to invest time into making this a reality. Especially since this addon is a temporary solution until Ember/Glimmer most likely does this out-of-the-box. If someone wants to implement it, it's fine, the skipped test is a good place to start. But in the end this is your addon, so feel free to decide however you like :) I like to keep skipped tests as a reminder to myself for very low-priority features, but that's just me ;)